File: src/EnvironmentMonitor.Domain/Enums/CommunicationChannels.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EnvironmentMonitor.Domain.Enums
{
    public enum CommunicationChannels
    {
        [Description("IoT Hub")]
        IotHub = 0,
        [Description("Rest API")]
        RestApi = 1
    }
}

File: src/EnvironmentMonitor.ReactUi/environment-monitor/src/enums/communicationChannels.ts
export enum CommunicationChannels {
  IotHub = 0,
  RestApi = 1,
  Ilmatieteenlaitos = 2,
}

export const getCommunicationChannelDisplayName = (
  channelId: CommunicationChannels,
): string => {
  switch (channelId) {
    case CommunicationChannels.IotHub:
      return "IoT Hub";
    case CommunicationChannels.RestApi:
      return "Rest API";
    case CommunicationChannels.Ilmatieteenlaitos:
      return "Ilmatieteenlaitos API";
    default:
      return "Unknown";
  }
};
